test/unix/main.ml

```diff
@@ -273,6 +273,28 @@ module Readonly = struct
     Index.close rw;
     Index.close ro
 
+  (* check that the ro instance is "snapshot isolated", e.g. it can read old
+     values, even when rw overwrites them. *)
+  let readonly_snapshot () =
+    let Context.{ rw; clone; tbl } = Context.full_index () in
+    let ro = clone ~readonly:true () in
+    check_equivalence rw tbl;
+    check_equivalence ro tbl;
```

Comment:
`clone` and `full_index` should guarantee this (and if we want to test that, we can do it separately).

```suggestion
```

(We could also remove this from the test above.)

Reply:
I think it's good to be explicit with the pre-conditions in this test